### PR TITLE
Add line continuation support to REPL

### PR DESCRIPTION
**Pull request by holygits**
- Can now do multi-line things in REPL
- General clean up

Also I'm not sure if the prompt should be `>>> ` or `>>>>> ` as the latter would need 5 dots for continued lines i.e. `..... `, not sure that looks as nice as `...`.

Tested the following inputs
```
>>> def greeting(who):
...    print("hello " + who)
>>> greeting("batman")

>>> class Person:
...    def __init__(self, name):
...        self.name = name
>>> Person('batman')
```

Some write up from cpython on this topic is [here](https://github.com/python/cpython/blob/67a93b3a0b3814e97ef9d077b21325fc8ce351b2/Doc/faq/extending.rst#how-do-i-tell-incomplete-input-from-invalid-input)

### CODE REVIEW

**Comment by cthulahoops** [comment]
The prompt should be ">>>>>" by default. This continues the pattern: Cpython uses 3, PyPy uses 4, we use 5.

The current code supports ```sys.ps1``` and ```sys.ps2```, this change remove that. 

I don't think checking the last character is sufficient to spot continuation lines, eg:

```
(1
+ 2)
```

We need to run the parser, and get it to report incomplete input. (This used to work but got broken at some point.)

File: src/main.rs [hunk 1]
```diff
@@ -153,58 +153,37 @@ fn run_shell(vm: &mut VirtualMachine) -> PyResult {
...
-                            }
-                            Err(msg) => panic!("Error: {:?}", msg),
-                        }
+                repl.add_history_entry(line.trim_end().as_ref());
+
+                // Check for continuation char(s)
+                match line.chars().last().unwrap_or('\0') {
+                    ':' | '\\' => {
+                        // Switch to nested prompt & get more input
+                        prompt = "... ";
```
Comment (windelbouwman) [inline src/main.rs]: For the promp sys.ps2 should be used here, so the prompt is changeable.

File: parser/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+use std::fmt;
+
+#[derive(Debug)]
+pub enum ParseError {
+    // Parser encountered an unexpected end of input
+    EOF,
+    // Parser encountered an extra token
+    ExtraToken,
+    // Parser encountered an invalid token
+    InvalidToken,
```
Comment (windelbouwman) [inline parser/src/error.rs]: This is handy! Can you add location information as well? So then we know where the error occurred.

File: vm/src/compile.rs [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-//!
 //! Take an AST and transform it into bytecode
 //!
 //! Inspirational code:
 //!   https://github.com/python/cpython/blob/master/Python/compile.c
 //!   https://github.com/micropython/micropython/blob/master/py/compile.c
 
 use super::bytecode::{self, CallType, CodeObject, Instruction};
-use super::pyobject::{PyObject, PyObjectPayload, PyResult};
-use super::vm::VirtualMachine;
```
Comment (windelbouwman) [inline vm/src/compile.rs]: The removal of the VirtualMachine here is very good!

File: vm/src/error.rs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+use rustpython_parser::ast;
+use rustpython_parser::error::ParseError;
+
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug)]
+pub enum CompileError {
+    // Cannot store value in target
+    Assign(ast::Expression),
```
Comment (windelbouwman) [inline vm/src/error.rs]: Instead of storing the expression here, insert its location only, then we do not require the clone attribute on Expression.

**Comment by windelbouwman** [comment]
@holygits you did some very good work on introducing proper error types instead of strings in the parser. Could you put those into a fresh pull request? The changes are unrelated to the REPL and otherwise there are many conflicting files in this pull request.

File: vm/src/error.rs (hunk 4 above)
Comment (holygits) [reply vm/src/error.rs]: Ah nice one. Will do

**Comment by holygits** [comment]
Moved error scaffolding to #464

**Pull request merged**